src/php/Command/Format/PhelPathFilter.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Phel\Command\Format;
+
+use RecursiveDirectoryIterator;
+use RecursiveIteratorIterator;
+
+final class PhelPathFilter implements PathFilterInterface
+{
+    private const PHEL_EXTENSION = 'phel';
+
+    /**
+     * @param list<string> $paths
+     *
+     * @return list<string> The recursively unique valid paths to be formatted
+     */
+    public function filterPaths(array $paths): array
+    {
+        $returnPaths = [];
+
+        foreach ($paths as $path) {
+            if (is_file($path)) {
+                if ($this->hasPhelExtension($path)) {
+                    $returnPaths[] = $path;
+                }
+                continue;
+            }
+
+            foreach ($this->createRecursiveIterator($path) as $fileInfo) {
```

Comment:
Is it save to say if it is not a file it is a directory?

* `is_file`: Returns true if the filename exists and is a regular file, false otherwise.
* `is_dir`: Returns true if the filename exists and is a directory, false otherwise. 

If I add a path that doesn't exists, what will happen?

Reply:
I didn't test that case, you are right. I updated the condition to this logic:
```php
if (is_file($path)) {
    // ...
} elseif (is_dir($path)) {
    // ...
}
```